From 47b4bfc35023fe4cf2233b4090954f2a1375c544 Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Tue, 24 Apr 2018 10:14:12 +0800
Subject: [PATCH 1/4] allow app to access priv_app_tmpfs

From kernel 4.14, untrusted app need to own read permission
of priv_app_tmpfs to access contacts provied by ContactProvider.
Cts runs in untrusted_app domain, it should to own read permission
to priv_app_tmpfs to complete some test cases.

Jira: None
Test: None

Change-Id: I60bd7b72db38f1c9c1eea9da88b04ebdb9e0c79f
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 private/app.te      | 3 +++
 private/priv_app.te | 1 +
 2 files changed, 4 insertions(+)

diff --git a/private/app.te b/private/app.te
index 9251ed9..da31df1 100644
--- a/private/app.te
+++ b/private/app.te
@@ -11,6 +11,9 @@
 # Read system properties managed by zygote.
 allow appdomain zygote_tmpfs:file read;
 
+# Read contacts Managed by Contact
+allow appdomain priv_app_tmpfs:file read;
+
 # WebView and other application-specific JIT compilers
 allow appdomain self:process execmem;
 
diff --git a/private/priv_app.te b/private/priv_app.te
index 60fb411..6ed3fe7 100644
--- a/private/priv_app.te
+++ b/private/priv_app.te
@@ -4,6 +4,7 @@
 
 typeattribute priv_app coredomain;
 app_domain(priv_app)
+typeattribute priv_app_tmpfs mlstrustedobject;
 
 # Access the network.
 net_domain(priv_app)
-- 
1.9.1

